Source/santactl/Commands/SNTCommandStatus.m

```diff
@@ -135,9 +135,18 @@ - (void)runWithArguments:(NSArray *)arguments {
 
   __block BOOL pushNotifications = NO;
   if ([[SNTConfigurator configurator] syncBaseURL]) {
-    [rop pushNotifications:^(BOOL response) {
-      pushNotifications = response;
-    }];
+    // The request to santad to discover whether push notifications are enabled
+    // makes a call to santasyncservice. If it's unavailable the call can hang
+    // so we run the request asynchronously with a semaphore timer; if we have
+    // no response within 2s, give up and move on.
+    dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+    dispatch_async(dispatch_get_global_queue(0, 0), ^{
```

Comment:
nit: Would recommend setting `QOS_CLASS_DEFAULT` instead of `QOS_CLASS_UNSPECIFIED`

Reply:
Done, except I went with `QOS_CLASS_USER_INITIATED` because I think that's arguably more correct for this case.